src/main/java/graphql/annotations/processor/GraphQLAnnotations.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.processor;
+
+import graphql.annotations.processor.exceptions.GraphQLAnnotationsException;
+import graphql.annotations.processor.retrievers.GraphQLObjectHandler;
+import graphql.annotations.annotationTypes.GraphQLName;
+import graphql.annotations.annotationTypes.GraphQLTypeExtension;
+import graphql.annotations.processor.graphQLProcessors.GraphQLAnnotationsProcessor;
+import graphql.annotations.processor.graphQLProcessors.GraphQLInputProcessor;
+import graphql.annotations.processor.graphQLProcessors.GraphQLOutputProcessor;
+import graphql.annotations.processor.typeFunctions.DefaultTypeFunction;
+import graphql.annotations.processor.typeFunctions.TypeFunction;
+import graphql.annotations.processor.retrievers.GraphQLObjectInfoRetriever;
+import graphql.relay.Relay;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLObjectType;
+import org.osgi.service.component.annotations.Component;
+
+import java.util.*;
+
+import static graphql.annotations.processor.util.NamingKit.toGraphqlName;
+
+/**
+ * A utility class for extracting GraphQL data structures from annotated
+ * elements.
+ */
+@Component
+public class GraphQLAnnotations implements GraphQLAnnotationsProcessor {
+
+    private GraphQLObjectHandler graphQLObjectHandler;
+    private ProcessingElementsContainer container;
```

Comment:
I really like the fact that the container is now separated from the GraphQLAnnotations class - it's actually an issue I was trying to solve, as I sometimes need to recreate a full new container/registry, but did not want to recreate a new GraphQLAnnotations instance (which is not supposed to be done when using OSGi, as GraphQLAnnotations is itself a Component - one single instance is created by the OSGi framework, and shared with everybody). However to make it completely decoupled it would be needed that the container is not a field of GraphQLAnnotations, but rather pass the container object as parameter when needed (mainly, when calling object() and other registration methods  ).
initializeContainer could then be made public ( maybe renamed "createNewContainer" ), instead of being called by the constructor, so that the user can get a brand new container when needed. The GraphQLAnnotations will not contain any state and the same instance can be used in different contexts, with different container.

Reply:
I dont want to make users pass container in these multiple methods.. Its breaking the api and im not sure that they will use the same instance in the different calls (and its important to use the same instance because the register etc methods are changing it).

I will add a public setContainer method that allows you to set to a new container without the need of recreating GraphQLAnnotations class

Reply:
A setContainer method will be fine in my case (or even resetContainer which would just create and set a new container), but I think it's still quite limited - as GraphQLAnnotations contain a state, it won't be possible to share the same instance between different applications, which could be a common use case in OSGi.  But I understand your point about the API - in that case it may be better to have a GraphQLAnnotationsFactory as an OSGi component, instead of the GraphQLAnnotations.

Reply:
Actually when using OSGi we could now completely forget about GraphQLAnnotations class and rather use container and GraphQLObjectHandler directly. That may makes more sense, so GraphQLAnnotations provides a simple shortcut to non OSGi environments.